### ISSUE TITLE
ZenPublisher v4.5.0.20130526 breaks Zenphoto version 1.4.5.1

### ISSUE THREAD

**Issue by igchy** [philbertphotos/Zenphoto-Lightroom-Publisher#20]
As soon I enable ZenPublisher plugin in Zenphoto I see blank pages on image uploads and when I submit and Zenphoto settings changes.

Checked on Zenphoto forum, looks like this was already reported for 1.4.5 version of Zenphoto.

Please see the original topic on zenphoto support forum:

http://www.zenphoto.org/support/topic.php?id=175383

I'm setting up testing environment so I'll try to contribute to this great project.

**Comment by jasonized** [comment]
There's something possibly going on in the install.   I moved zenPublisher out of the plugins, re-installed, then put zenPublisher back.   It appears to be working for me now (well, except for the LR5 bug, which I worked around).

If in doubt, re-install zenphoto (and of course, run setup, etc.)  That seems to have fixed it for me.

Hmm.. I should probably mention I ended up re-installing zenphoto twice;  first time I tried it, strange things happened.  The second time I re-installed I did not move zenPublisher and everything installed properly.  I did note that I had problems originally which stemmed from old code left in the zp-core area;  deleting that and then installing seems to have fixed everything else.

**Comment by igchy** [comment]
I just re-installed Zenphoto manually from  the scratch.
By "manually" I mean not using Softaculous scripts provided by my web-hoster, but downloading code from zenphoto.org, running setup and creating database/user.
And I see all the old problems.

Could you explain how exactly you re-installed zenphoto?
And what do you mean by "did not move zenPublisher"?
Does it mean you'd put ZenPublisher files in plugins directory before running setup.php?

Just noticed this warning all other zenphoto debug log:

WARNING: Cannot modify header information - headers already sent by (output started at /path_to_zenphoto/plugins/ZenPublisher.php:103) in /path_to_zenphoto/zp-core/admin-functions.php on line 110
header called from printAdminHeader (admin-functions.php [110])
from admin-logs.php [69]

**Comment by jasonized** [comment]
Okay, what I did turns out may or may not help;  I just tried to update 
my working copy (I did all my testing on a test bed copy of my current 
site), and while the testbed now functions, when I did the updates on my 
real site, I got errors saying "too many redirections".  Sigh.

But, if it helps:

```
I first saw the message saying zenpublisher was causing problems 
```

with the update.  So what I did was to move the zenpublisher files out 
of my tree (just moved them into a holding directory above the main web 
directory, so they couldn't be seen).
   Then I downloaded, unzipped and copied (cpio) all the files into 
place, and ran the install scripts.
   It looked okay, more or less (no crashes, theme looked a bit strange).
   I then moved the zenpublisher files back into the plugins directory. 
   Checked and found a bug caused by LR5, and worked around it.
   I noticed I had an extra blank album (from previous testing things), 
so deleted it from the admin page.
   My test site went to a white page.  No error messages, or debug logs.
   I copied in all of the downloaded files again, and re-ran setup. My 
site started working again.   I was able to create/delete albums, and 
only had some theme weirdnesses to fix.

After quite a bit of emailing and work, I've got the test site working 
fairly well now.

  So I decided to update my real site.  Before I did that, however, I 
made a full tar file of my current system.
Good thing I did.
I ran the updates, and my main site went blank.  I re-ran the updates, 
and now my site gave me -part- of the main page and stopped.  Errors in 
the log file said too many redirections.

So I wiped it out and reinstalled my old setup, and it's working.  I 
have no idea what is going on with the redirects, or why the test site 
isn't getting them.   I'll probably try again this weekend, if I have 
time, and see if I can track it down.  So I don't think it is an issue 
with zenpublisher so much as just weirdness with updating from a 
previous version problem, since the symptoms change based on the 
previous setup.

Not the most fun update I've done.

**Comment by igchy** [comment]
thanks for spending time trying to help me here.
in my case both zenphoto and zendpublisher were installed from scratch, there was no old version and updates/upgrades.
i'll try to spin up some VM on my computer during a weekend and play with it in environment where i have full access.

**Comment by jasonized** [comment]
Yeah, not the easiest thing to diagnose, even with full access.  I just 
tried my main site again, and this time I'm getting internal server 
errors whenever I hit anything besides the main page.

Arggh.

Jason

**Comment by jasonized** [comment]
Here's a really weird one.   I finally got my real site working (mostly: 
slideshow is still off to one side, for some reason). Wiped out, 
reinstalled, wiped out, copied in, kept getting messages saying "there's 
a problem with index.php, setup.php".. weird.   One thing that might 
have helped, I ended up changing the dir permissions to 777 and wiping 
out .htaccess.  I know, I updated it, but for some reason it wanted to 
do it itself...   After that, and various other copy incantations, it 
finally installed.   The site kept doing random things for awhile (log 
file stated index.php was missing, for example, but refreshing the page 
worked, so what, it wandered out for coffee?).  But seems to be working 
for now.

  But here's the kicker for LR:  Don't go into the plugins and enable 
ZenPublisher.   If you do that, you start getting error messages. But it 
doesn't seem to matter, because even with the plugin disabled, LR still 
publishes just fine...  ?

Gotta be the worst update problems I've had yet with Zenphoto.. and 
considering there are _always_ issues updating, that's going some. Oh 
well.  It mostly works, just need to find out why slideshow doesn't 
follow anything else now..

Cheers..

Jason

**Comment by igchy** [comment]
ok, my 2 cents:
just completed following:
1) deployed ubuntu server 12.04.2 LTS in a new VM under MS hyper-v;
2) installed default ubuntu LAMP stack, which is apache 2.2.1/mysql 5.5._/php 5.3._
3) installed zenphoto 1.4.5.1 + ZenPublisher 4.5.0.20130529
4) installed adobe lightroom 5.2 rc on 8.1 prevew

I DO NOT see blank pages or any other artifacts.
It works perfectly fine in hyper-v vm on my laptop.

Please let me know how can I reproduce any of issue you've seen on your instances.

That makes me think something is broken on my web hosting.
